utils/check-quickstart-slug.ts

```diff
@@ -0,0 +1,207 @@
+import * as path from 'path';
+
+import {
+  findMainQuickstartConfigFiles,
+  readYamlFile,
+  removeRepoPathPrefix,
+} from './helpers';
+
+import fetch from 'node-fetch';
+import { readFileSync } from 'fs';
+import { writeFile } from 'fs/promises';
+
+export interface QuickstartConfig {
+  id: string;
+  name: string;
+  description: string;
+  summary: string;
+  logo: string;
+  level: string;
+  authors: Array<string>;
+  title: string;
+  documentation: Array<string>;
+  keywords: Array<string>;
+  installPlans: Array<string>;
+}
+
+interface YamlFile {
+  path: string;
+  contents: QuickstartConfig[];
+}
+
+/**
+ * Helper function that slugifies a string for URL usage.
+ * @param str string to slugify
+ * @returns slugified string
+ */
+const slugify = (str: string): string =>
+  str
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, '-')
+    .replace(/-+/, '-')
+    .replace(/[^a-z0-9-]/g, '');
+
+/**
+ * Helper function to compare two slugs
+ * @param first slug to compare
+ * @param second slug to compare
+ * @returns Boolean results of slugs equality
+ */
+const isSlugSame = (first: string, second: string): boolean => first === second;
+
+/**
+ * Function replaces first regex match and replaces name with title
+ * @param rawConfig - raw buffer from fs.readFileSync
+ * @param title - title of quickstart from config
+ * @returns
+ */
+const fixConfig = (rawConfig: string, title: string): string => {
+  return rawConfig.replace(/^name:.*$/m, `name: ${slugify(title)}`);
+};
+
+/**
+ * Logging function for logging information
+ * @param info Tuple array containing [similars, diffs]
+ */
+const printInfo = (results: Array<string>): void => {
+  console.log('Number of failed quickstart URLs:', results.length);
+  console.log('Failed quickstart URLs:', results);
+};
+
+/**
+ *
+ * @param diffs Array of tuples containing the relative path in quickstart repo
+ * and contains content inside `config.yaml` where `name` and `title` differ.
+ * @returns Promise of failed quickstart paths.
+ */
+export const checkUrlResponse = (
+  diffs: Array<[string, QuickstartConfig]>
+): Promise<string[]> => {
+  const SITE_URL: string = 'https://newrelic.com/instant-observability';
+
+  const fetchResults: Promise<string>[] = diffs.map(async (tuple) => {
+    const [quickstartPath, contents] = tuple;
+    const resp = await fetch(`${SITE_URL}/${contents.name}/${contents.id}`);
+    if (!resp.ok || resp.status === 404) {
+      return quickstartPath;
+    }
+  });
+  return Promise.all(fetchResults);
+};
```

Comment:
This function is for verifying `contents.name` is the correct URL for the quickstart. (soon to be `contents.slug`)

Reply:
Also to note: I left the script in the PR for code review purposes. We do not have to check it into the repo's `main`